app/src/test/java/org/simple/clinic/settings/changelanguage/ChangeLanguageInitTest.kt

```diff
@@ -0,0 +1,74 @@
+package org.simple.clinic.settings.changelanguage
+
+import com.spotify.mobius.test.FirstMatchers.hasEffects
+import com.spotify.mobius.test.FirstMatchers.hasModel
+import com.spotify.mobius.test.FirstMatchers.hasNoEffects
+import com.spotify.mobius.test.InitSpec
+import com.spotify.mobius.test.InitSpec.assertThatFirst
+import org.junit.Test
+import org.simple.clinic.settings.Language
+
+class ChangeLanguageInitTest {
+
+  private val spec = InitSpec<ChangeLanguageModel, ChangeLanguageEffect>(ChangeLanguageLogic::init)
+  private val defaultModel = ChangeLanguageModel.FETCHING_LANGUAGES
+
+  @Test
+  fun `the load current language effect and load supported languages must be emitted when the model is created`() {
+    spec
+        .whenInit(defaultModel)
+        .then(assertThatFirst(
+            hasModel(defaultModel),
+            hasEffects(LoadCurrentSelectedLanguageEffect, LoadSupportedLanguagesEffect)
+        ))
+  }
+
+  @Test
+  fun `when the model is restored with the current language, the load supported languages effect must be emitted`() {
+    val language = Language(displayName = "English", languageCode = "en_IN")
+    val model = defaultModel.withCurrentLanguage(language)
+
+    spec
+        .whenInit(model)
+        .then(assertThatFirst(
+            hasModel(model),
+            hasEffects(LoadSupportedLanguagesEffect as ChangeLanguageEffect)
+        ))
+  }
+
+  @Test
+  fun `when the model is restored with the supported languages, the load current language effect must be emitted`() {
```

Comment:
`when the model is restored without the current language, then load current language effect must be emitted`?